### PR TITLE
Add pootle_fs state ob

### PR DESCRIPTION
**Pull request by phlax**
Add a state class for managing project fs, and some unit tests...

### CODE REVIEW

File: pootle/apps/pootle_fs/state.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    @property
+    def store(self):
+        return self.kwargs.get("store")
+
+    def __gt__(self, other):
+        if isinstance(other, self.__class__):
+            return self.pootle_path > other.pootle_path
+        return object.__gt__(other)
+
+    def __eq__(self, other):
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/state.py]: is it necessary if we use method of super class only?

File: pootle/apps/pootle_fs/state.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+from collections import OrderedDict
+
+from django.utils.functional import cached_property
+from django.utils.lru_cache import lru_cache
+
+from pootle.core.state import ItemState, State
+from pootle_store.models import FILE_WINS, POOTLE_WINS
+
+from .resources import FSProjectStateResources
+
```
Comment (unho) [inline pootle/apps/pootle_fs/state.py]: Add extra blank line.

File: pootle/apps/pootle_fs/state.py [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    "title": "Untracked Stores",
+    "description": "Newly created Stores in Pootle"}
+FS_STATE["pootle_staged"] = {
+    "title": "Added in Pootle",
+    "description": (
+        "Stores that have been added in Pootle and are now being tracked")}
+FS_STATE["pootle_removed"] = {
+    "title": "Removed from Pootle",
+    "description": "Stores that have been removed from Pootle"}
+FS_STATE["fs_staged"] = {
```
Comment (unho) [inline pootle/apps/pootle_fs/state.py]: Nit: Use the same order as for `pootle_`, so `*_ahead`, `*_untracked`, `*_staged`, `*_removed`.

File: pootle/apps/pootle_fs/state.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+        return self.resources.synced.exclude(pootle_path__in=has_changes)
+
+    @property
+    def state_fs_staged(self):
+        staged = (
+            self.resources.unsynced
+                          .filter(path__in=self.resources.found_file_paths)
+                          .exclude(resolve_conflict=POOTLE_WINS)
+            | self.resources.synced
+                            .filter(store__isnull=True)
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/state.py]: `.filter(Q(store__isnull=True)|Q(store__obsolete=True))`

File: pootle/apps/pootle_fs/state.py [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+                or store_fs.resolve_conflict == POOTLE_WINS)
+            if pootle_ahead:
+                yield dict(store_fs=store_fs)
+
+    @property
+    def state_pootle_staged(self):
+        staged = (
+            self.resources.unsynced
+                          .exclude(resolve_conflict=FILE_WINS)
+                          .exclude(store__isnull=True)
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/state.py]:
```
.exclude(store__isnull=True)
.exclude(store__obsolete=True)
```

File: pootle/apps/pootle_fs/state.py [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+            | self.resources.synced
+                            .exclude(store__isnull=True)
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/state.py]:
```
.exclude(store__obsolete=True)
```

File: pootle/apps/pootle_fs/state.py [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+                            .exclude(path__in=self.resources.found_file_paths)
+                            .filter(resolve_conflict=POOTLE_WINS))
+        for store_fs in staged.iterator():
+            store_fs.project = self.project
+            yield dict(store_fs=store_fs)
+
+    @property
+    def state_both_removed(self):
+        removed = self.resources.synced.filter(
+            store__isnull=True).exclude(path__in=self.resources.found_file_paths)
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/state.py]:
```
filter(Q(store__obsolete=True)|Q(store__isnull=True))
```

File: pootle/apps/pootle_fs/state.py [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+            store_fs.project = self.project
+            yield dict(store_fs=store_fs)
+
+    @property
+    def state_pootle_removed(self):
+        synced = (
+            self.resources.synced
+                          .exclude(resolve_conflict=FILE_WINS)
+                          .filter(path__in=self.resources.found_file_paths)
+                          .filter(store__isnull=True))
```
Comment (ta2-1) [inline pootle/apps/pootle_fs/state.py]: filter(Q(store__obsolete=True)|Q(store__isnull=True))

File: pootle/apps/pootle_fs/state.py (hunk 7 above)
Comment (phlax) [inline pootle/apps/pootle_fs/state.py]: already done locally 8/

File: pootle/apps/pootle_fs/state.py (hunk 2 above)
Comment (unho) [inline pootle/apps/pootle_fs/state.py]: Addressed.

File: pootle/apps/pootle_fs/state.py (hunk 5 above)
Comment (unho) [inline pootle/apps/pootle_fs/state.py]: Addressed.

**Comment by unho** [comment]
lgtm

**Pull request merged**